### ISSUE TITLE
How to incorporate oracledb with webdriver flow control mechanism?

### ISSUE THREAD

**Issue by vivekbrahmadandi** [oracle/node-oracledb#305]
Hi,
I am new to oracledb module. Would you help for the using in protracotor jasmine tests

I can see another issue for the same type 
https://github.com/oracle/node-oracledb/issues/110

**Comment by cjbj** [comment]
Have you got a specific question about something you've tried?

**Comment by vivekbrahmadandi** [comment]
I am working to develop a reusable piece of code to connect data base and execute queries and return the result ,in simple we need a wraper module for the above operations on oracle db , which is used to incorporated in our framework.

**Comment by vivekbrahmadandi** [comment]
If we can find any example integrations with protractor , it would be great to for us for the reference.I didnt find any online resources intregration of node oracle db using protractor.

**Comment by cjbj** [comment]
@vivekbrahmadandi are there any examples you can share?

**Comment by isian8814** [comment]
@cjbj 
@vivekbrahmadandi  Have you found a solution? I have the same questions.

I want to use a single connection and run multiple queries.
After the queries are finished, then I want to release the connection.
However I got the error of 
ErrorMsg: NJS-032: connection cannot be released because a database call is in progress

fit('test db connection', function () {
        var controlFlow = protractor.promise.controlFlow();
        controlFlow.wait(
            dbUtility.getDBConnection().then(function (connection) {
                dbUtility.executeQuery(connection);
                dbUtility.executeQuery(connection);
                dbUtility.releaseDBConnection(connection);
            })
        );
    });

 getDBConnection : function () {
        var defer = protractor.promise.defer();
        connectDatabase().then(function (connection) {
            databaseConnection = connection;
            defer.fulfill(databaseConnection);
        })
        return defer.promise;
    },
    
    executeQuery : function (databaseConnection) {
        var defer = protractor.promise.defer();
        databaseConnection.execute(
            "select sysdate from dual ",
            [],
            {
                resultSet: true, // return a result set.  Default is false
                prefetchRows: 25 // the prefetch size can be set for each query
            },
            function(err, result)
            {
                if (err) {
                    console.error(err.message);
                    doRelease(databaseConnection);
                    return;
                }
                console.log(result.resultSet);
                defer.fulfill(result.resultSet);
            });
        return defer.promise;
    }

 releaseDBConnection : function (databaseConnection) {
        var defer = protractor.promise.defer();
        doRelease(databaseConnection);
        defer.fulfill();
        return defer.promise;
    }

**Comment by cjbj** [comment]
This sequence looks suspicious:
```
      dbUtility.executeQuery(connection);
      dbUtility.executeQuery(connection);
      dbUtility.releaseDBConnection(connection);
```

You need to make them run sequentially, not in parallel.

**Comment by cjbj** [comment]
Closing due to lack of feedback.
